tests/test_tags.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2020 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+import requests
+from huggingface_hub.hf_api import HfApi
+from huggingface_hub.utils.tags import (
+    AttributeDictionary,
+    DatasetTags,
+    GeneralTags,
+    ModelTags,
+)
+
+from .testing_utils import with_production_testing
+
+
+class AttributeDictionaryCommonTest(unittest.TestCase):
+    _attrdict = AttributeDictionary()
+
+
+class AttributeDictionaryTest(AttributeDictionaryCommonTest):
+    def setUp(self):
+        self._attrdict.clear()
+
+    def test_adding_item(self):
+        self._attrdict["itemA"] = 2
+        self.assertEqual(self._attrdict.itemA, 2)
+        self.assertEqual(self._attrdict["itemA"], 2)
+        # We should be able to both set a property and a key
+        self._attrdict.itemB = 3
+        self.assertEqual(self._attrdict.itemB, 3)
+        self.assertEqual(self._attrdict["itemB"], 3)
+
+    def test_removing_item(self):
+        self._attrdict["itemA"] = 2
+        self._attrdict.itemB = 3
```

Comment:
nit: maybe move this one before line 53 so the assignment is coupled with the test, just as done in previous test.

Reply:
Nice readability call. Also did so for another one too :)